src/imgui/ImGuiMachine.cc

```diff
@@ -46,9 +46,63 @@ ImGuiMachine::ImGuiMachine(ImGuiManager& manager_)
 	, setupFileList("setup", Reactor::SETUP_EXTENSION, Reactor::SETUP_DIR)
 	, confirmDialog("Confirm##setup")
 {
-	setupFileList.selectAction = [&](const FileListWidget::Entry& entry) {
+
+	setupFileList.drawAction = [&] {
+		im::Table("table", 2, ImGuiTableFlags_BordersInnerV, [&]{
+			if (ImGui::TableNextColumn()) {
+				setupFileList.drawTable();
+			}
+			if (ImGui::TableNextColumn()) {
+				if (previewSetup.motherBoard) {
+					showSetupOverview(*previewSetup.motherBoard);
+					ImGui::Separator();
+					if (ImGui::Button("Run!")) {
+						manager.executeDelayed(makeTclList("setup", previewSetup.displayName));
+					}
+				} else {
+					if (previewSetup.lastExceptionMessage.empty()) {
+						ImGui::TextUnformatted("Nothing to preview...");
```

Comment:
Detail: maybe it's nicer to show this text grayed-out? I think we do this in some cases already, but maybe not everywhere?

Reply:
Maybe, but I don't think it's very useful here. Feel free to tweak if you prefer it.